internal/cli/atlas_measurements_process.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mongocli/internal/description"
+	"github.com/mongodb/mongocli/internal/flags"
+	"github.com/mongodb/mongocli/internal/json"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type atlasMeasurementsProcessOpts struct {
+	*globalOpts
+	pageNum         int
+	itemsPerPage    int
+	host            string
+	port            int
+	granularity     string
+	period          string
+	start           string
+	end             string
+	measurementType string
+	store           store.ProcessMeasurementLister
+}
+
+func (opts *atlasMeasurementsProcessOpts) init() error {
+	if opts.ProjectID() == "" {
+		return errMissingProjectID
+	}
+
+	var err error
+	opts.store, err = store.New()
+	return err
+}
+
+func (opts *atlasMeasurementsProcessOpts) Run() error {
+	listOpts := opts.newProcessMeasurementListOptions()
+	result, err := opts.store.ListProcessMeasurements(opts.ProjectID(), opts.host, opts.port, listOpts)
+
+	if err != nil {
+		return err
+	}
+
+	return json.PrettyPrint(result)
+}
+
+func (opts *atlasMeasurementsProcessOpts) getHostNameAndPort(hostInfo string) (string, int, error) {
+	host := strings.SplitN(hostInfo, ":", -1)
+	if len(host) != 2 {
+		return "", 0, fmt.Errorf("expected hostname:port, got %s", host)
+	}
+
+	port, err := strconv.Atoi(host[1])
+	if err != nil {
+		return "", 0, err
```

Comment:
same as before, could we improve on the error and show something like, invalid port number, got %s